server/s3_copy_object_action.cc

```diff
@@ -82,6 +84,145 @@ void S3CopyObjectAction::setup_steps() {
   ACTION_TASK_ADD(S3CopyObjectAction::send_response_to_s3_client, this);
 }
 
+void S3CopyObjectAction::get_source_bucket_and_object() {
+  s3_log(S3_LOG_DEBUG, request_id, "Entering\n");
+  std::string source = request->get_copy_object_source();
+  size_t separator_pos = source.find("/");
+  if (separator_pos != std::string::npos) {
+    source_bucket_name = source.substr(0, separator_pos);
+    source_object_name = source.substr(separator_pos + 1);
+  }
```

Comment:
Should we handle incorrect source "path"?

Reply:
Yes, it is handled in validate_copyobject_request(), from where this function is being called.